File: pbweb/pbweb-jar/src/main/java/it/csi/pbandi/pbweb/business/service/GestioneIntegrazioniService.java
package it.csi.pbandi.pbweb.business.service;

import it.csi.pbandi.pbweb.dto.IntegrazioneRendicontazioneDTO;
import it.csi.pbandi.pbweb.dto.IntegrazioneRevocaDTO;
import it.csi.pbandi.pbweb.dto.rendicontazione.AllegatoDTO;
import it.csi.pbandi.pbweb.integration.vo.AllegatiDichiarazioneSpesaVO;
import it.csi.pbandi.pbweb.integration.vo.VisualizzaIntegrazioniVO;
import org.springframework.web.bind.annotation.RequestBody;

import javax.servlet.http.HttpServletRequest;
import javax.ws.rs.*;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import java.security.InvalidParameterException;
import java.util.List;

@Path("/gestioneIntegrazioni")
public interface GestioneIntegrazioniService {
    @GET
    @Path("/getAbilitaInviaIntegr")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAbilitaInviaIntegr(
            @DefaultValue("0") @QueryParam("idRichIntegrazione") int idRichIntegrazione,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getAllegatiIstruttoreRevoche")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiIstruttoreRevoche(
            @DefaultValue("0") @QueryParam("idRichIntegraz") int idRichIntegraz,
            @Context HttpServletRequest req) throws Exception;
    @GET
    @Path("/getAllegati")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegati(
            @DefaultValue("0") @QueryParam("idRichIntegraz") int idRichIntegraz,
            @Context HttpServletRequest req) throws Exception;
    @GET
    @Path("/getLetteraIstruttore")
    @Produces(MediaType.APPLICATION_JSON)
    Response getLetteraIstruttore(
            @DefaultValue("0") @QueryParam("idRichIntegraz") int idRichIntegraz,
            @Context HttpServletRequest req) throws Exception;

    @POST
    @Path("/salvaUploadLetteraAllegatoIntegr/{idRichIntegraz}")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response salvaUploadLetteraAllegatoIntegr(
            @PathParam("idRichIntegraz") Long idRichIntegraz,
            List<VisualizzaIntegrazioniVO> allegati,
            @Context HttpServletRequest req) throws Exception;

    @POST
    @Path("/updateRichIntegrazione")
    @Produces(MediaType.APPLICATION_JSON)
    Response updateRichIntegrazione(
            @Context HttpServletRequest req,
            @RequestBody VisualizzaIntegrazioniVO vc)
            throws Exception;

    @POST
    @Path("/deleteAllegato")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response deleteAllegato(
            VisualizzaIntegrazioniVO acv,
            @Context HttpServletRequest req
    ) throws Exception;

    @GET
    @Path("/getAllegatiDichSpesa")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiDichSpesa(
            @DefaultValue("0") @QueryParam("idDichSpesa") int idDichSpesa,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getDocumentiSpesaSospesi")
    @Produces(MediaType.APPLICATION_JSON)
    Response getDocumentiSpesaSospesi(
            @DefaultValue("0") @QueryParam("idProgetto") int idProgetto,
            @Context HttpServletRequest req) throws Exception;

    @POST
    @Path("salvaAllegatiGenerici")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response salvaAllegatiGenerici(
            @DefaultValue("0") @QueryParam("idDichiarazioneSpesa") Long idDichiarazioneSpesa,
            @DefaultValue("0") @QueryParam("idIntegrazioneSpesa") Long idIntegrazioneSpesa,
            List<AllegatiDichiarazioneSpesaVO> allegati,
            @Context HttpServletRequest req) throws Exception;

    //INTEGRAZIONE ALLA REVOCA
    @GET
    @Path("getIntegrazioniRevoca")
    @Produces(MediaType.APPLICATION_JSON)
    Response getIntegrazioniRevoca(
            @DefaultValue("0") @QueryParam("idProgetto") Long idProgetto,
            @DefaultValue("0") @QueryParam("idBandoLinea") Long idBandoLinea,
            @Context HttpServletRequest req) throws Exception;
    @GET
    @Path("getRegoleIntegrazione")
    @Produces(MediaType.APPLICATION_JSON)
    Response getRegoleIntegrazione(
            @DefaultValue("0") @QueryParam("idBandoLinea") Long idBandoLinea,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getRichProroga")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response getRichProroga(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @Context HttpServletRequest req) throws Exception;

    @POST
    @Path("/inserisciRichProroga/{idIntegrazione}")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response inserisciRichProroga(
            @PathParam("idIntegrazione") Long idIntegrazione,
            IntegrazioneRevocaDTO integrazioneRevocaDTO,
            @Context HttpServletRequest req) throws Exception;

    @POST
    @Path("/inviaIntegrazione/{idIntegrazione}")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response inviaIntegrazione(
            @PathParam("idIntegrazione") Long idIntegrazione,
            @Context HttpServletRequest req) throws Exception;

    //INTEGRAZIONE ALLA RENDICONTAZIONE
    @GET
    @Path("/getIntegrazioniRendicontazione")
    @Produces(MediaType.APPLICATION_JSON)
    Response getIntegrazioniRendicontazione(
            @DefaultValue("0") @QueryParam("idProgetto") Long idProgetto,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getRichProrogaRendicontazione")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response getRichProrogaRendicontazione(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @Context HttpServletRequest req) throws Exception;

    @POST
    @Path("/inserisciRichProrogaRendicontazione/{idIntegrazione}")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response inserisciRichProrogaRendicontazione(
            @PathParam("idIntegrazione") Long idIntegrazione,
            IntegrazioneRendicontazioneDTO integrazioneRendicontazioneDTO,
            @Context HttpServletRequest req) throws Exception;

    @POST
    @Path("/inviaIntegrazioneRendicontazione/{idIntegrazione}")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response inviaIntegrazioneRendicontazione(
            @PathParam("idIntegrazione") Long idIntegrazione,
            @Context HttpServletRequest req) throws Exception;

    //ALLEGATI
    @GET
    @Path("/getLetteraAccIntegrazRendicont")
    @Produces(MediaType.APPLICATION_JSON)
    Response getLetteraAccIntegrazRendicont(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getReportValidazione")
    @Produces(MediaType.APPLICATION_JSON)
    Response getReportValidazione(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getAllegatiDichiarazioneSpesa")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiDichiarazioneSpesa(
            @DefaultValue("0") @QueryParam("idDichiarazioneSpesa") Long idDichiarazioneSpesa,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getAllegatiIntegrazioneDS")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiIntegrazioneDS(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @DefaultValue("0") @QueryParam("idDichiarazioneSpesa") Long idDichiarazioneSpesa,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getAllegatiNuovaIntegrazioneDS")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiNuovaIntegrazioneDS(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @DefaultValue("0") @QueryParam("idDichiarazioneSpesa") Long idDichiarazioneSpesa,
            @Context HttpServletRequest req) throws Exception;

    //TAB Documenti di spesa sospesi
    @GET
    @Path("/getDocumentiDiSpesaSospesi")
    @Produces(MediaType.APPLICATION_JSON)
    Response getDocumentiDiSpesaSospesi(
            @DefaultValue("0") @QueryParam("idDichiarazioneSpesa") Long idDichiarazioneSpesa,
            @DefaultValue("0") @QueryParam("idProgetto") Long idProgetto,
            @Context HttpServletRequest req) throws Exception;
    @GET
    @Path("/getDocumentiDiSpesaIntegrati")
    @Produces(MediaType.APPLICATION_JSON)
    Response getDocumentiDiSpesaIntegrati(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @DefaultValue("0") @QueryParam("idDichiarazioneSpesa") Long idDichiarazioneSpesa,
            @DefaultValue("0") @QueryParam("idProgetto") Long idProgetto,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getAllegatiDocumentoSpesa")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiDocumentoSpesa(
            @DefaultValue("0") @QueryParam("idDichiarazioneSpesa") Long idDichiarazioneSpesa,
            @DefaultValue("0") @QueryParam("idDocumentoDiSpesa") Long idDocumentoSpesa,
            @Context HttpServletRequest req) throws Exception;
    @GET
    @Path("/getAllegatiIntegrazioneDocS")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiIntegrazioneDocS(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @DefaultValue("0") @QueryParam("idDichiarazioneSpesa") Long idDichiarazioneSpesa,
            @DefaultValue("0") @QueryParam("idDocumentoDiSpesa") Long idDocumentoSpesa,
            @Context HttpServletRequest req) throws Exception;
    @GET
    @Path("/getAllegatiNuovaIntegrazioneDocS")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiNuovaIntegrazioneDocS(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @DefaultValue("0") @QueryParam("idDocumentoDiSpesa") Long idDocumentoSpesa,
            @Context HttpServletRequest req) throws Exception;


    @GET
    @Path("/getQuietanze")
    @Produces(MediaType.APPLICATION_JSON)
    Response getQuietanze(
            @DefaultValue("0") @QueryParam("idDocumentoSpesa") Long idDocumentoSpesa,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getAllegatiQuietanza")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiQuietanza(
            @DefaultValue("0") @QueryParam("idDichiarazioneSpesa") Long idDichiarazioneSpesa,
            @DefaultValue("0") @QueryParam("idQuietanza") Long idQuietanza,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getAllegatiIntegrazioneQuietanza")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiIntegrazioneQuietanza(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @DefaultValue("0") @QueryParam("idDichiarazioneSpesa") Long idDichiarazioneSpesa,
            @DefaultValue("0") @QueryParam("idQuietanza") Long idQuietanza,
            @Context HttpServletRequest req) throws Exception;

    @GET
    @Path("/getAllegatiNuovaIntegrazioneQuietanza")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegatiNuovaIntegrazioneQuietanza(
            @DefaultValue("0") @QueryParam("idIntegrazione") Long idIntegrazione,
            @DefaultValue("0") @QueryParam("idQuietanza") Long idQuietanza,
            @Context HttpServletRequest req) throws Exception;

    //AGGIUNGI/RIMUOVI ALLEGATI
    @POST
    @Path("salvaAllegati")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response salvaAllegati(
            List<AllegatoDTO> allegati,
            @Context HttpServletRequest req) throws Exception;
    @DELETE
    @Path("/rimuoviAllegato")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response rimuoviAllegato(
            @DefaultValue("0") @QueryParam("idFileEntita") Long idFileEntita,
            @Context HttpServletRequest req) throws Exception;
    
    @GET
	@Path("allegatiDocSpesaQuietanzeDaInviare")
	@Produces(MediaType.APPLICATION_JSON)
	public Response getAllegatiDocSpesaQuietanzeDaInviare(@QueryParam("idIntegrazioneSpesa") Long idIntegrazioneSpesa,
			@Context HttpServletRequest req) throws InvalidParameterException, Exception;
}

File: pbwebwcl/src/app/gestione-integrazioni/services/gestione-integrazioni.service.ts
import { Observable } from 'rxjs';
import { DOCUMENT } from '@angular/common';
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Inject, Injectable } from '@angular/core';
import { ConfigService } from 'src/app/core/services/config.service';
import { IntegrazioniRevocaDTO } from '../commons/dto/integrazioniRevocaDTO';
import { AllegatiIntegrazioniVO } from '../commons/dto/allegatiIntegrazioniDTO';
import { IntegrazioniRendicontazioneDTO } from '../commons/dto/integrazioniRendicontazioneDTO';
import { AllegatiDichiarazioneSpesaVO } from '../commons/dto/allegatiDichiarazioneSpesaVO';
import { DocumentiSpesaSospesiVO } from '../commons/dto/DocumentiSpesaSospesiVO';
import { AllegatiControlliVO } from '../commons/dto/allegatiControlliDTO';
import { IntegrazioniControlliDTO } from '../commons/dto/integrazioniControlliDTO';
import { AllegatiPrecedentementeVO } from '../commons/allegatiPrecedentementeVO';
import { DocIntegrRendDTO } from '../commons/dto/doc-integr-rend-dto';
import { DocumentoDiSpesaSospesoDTO } from '../commons/dto/documento-di-spesa-sospeso-dto';
import { AllegatiDocSpesaQuietanzeDTO } from '../commons/dto/allegati-doc-spesa-quietanze-dto';
import { QuietanzaDTO } from '../commons/dto/quietanza-dto';

@Injectable()
export class GestioneIntegrazioniService {

    constructor(
        private configService: ConfigService,
        private http: HttpClient,
        @Inject(DOCUMENT) private document: any
    ) { }

    //RENDICONTAZIONE


    getIntegrazioniRendicontazione(idProgetto: number, idBandoLinea: number): Observable<Array<IntegrazioniRendicontazioneDTO>> {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getIntegrazioniRendicontazione';
        let params = new HttpParams().set('idProgetto', idProgetto.toString());
        return this.http.get<Array<IntegrazioniRendicontazioneDTO>>(url, { params: params });
    }

    getRegoleIntegrazione(idBandoLinea: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getRegoleIntegrazione';
        let params = new HttpParams().set('idBandoLinea', idBandoLinea.toString());
        return this.http.get<any>(url, { params: params });
    }


    getLetteraAccIntegrazRendicont(idIntegrazione: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getLetteraAccIntegrazRendicont';
        let params = new HttpParams().set('idIntegrazione', idIntegrazione.toString());
        return this.http.get<Array<DocIntegrRendDTO>>(url, { params: params });
    }

    getAllegatiIstruttoreRevoche(idRichIntegraz: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegatiIstruttoreRevoche';
        let params = new HttpParams().set('idRichIntegraz', idRichIntegraz.toString());
        return this.http.get<Array<AllegatiPrecedentementeVO>>(url, { params: params });
    }

    getAllegatiDichiarazioneSpesa(idDichiarazioneSpesa: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegatiDichiarazioneSpesa';
        let params = new HttpParams().set('idDichiarazioneSpesa', idDichiarazioneSpesa.toString());
        return this.http.get<Array<AllegatiPrecedentementeVO>>(url, { params: params });
    }

    getAllegatiIntegrazioneDS(idDichiarazioneSpesa: number, idIntegrazione: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegatiIntegrazioneDS';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString())
            .set('idDichiarazioneSpesa', idDichiarazioneSpesa.toString());
        return this.http.get<Array<AllegatiPrecedentementeVO>>(url, { params: params });
    }

    getAllegatiNuovaIntegrazioneDS(idDichiarazioneSpesa: number, idIntegrazione: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegatiNuovaIntegrazioneDS';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString())
            .set('idDichiarazioneSpesa', idDichiarazioneSpesa.toString());
        return this.http.get<Array<AllegatiDichiarazioneSpesaVO>>(url, { params: params });
    }

    getDocumentiDiSpesaSospesi(idDichiarazioneSpesa: number, idProgetto: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getDocumentiDiSpesaSospesi';
        let params = new HttpParams()
            .set('idDichiarazioneSpesa', idDichiarazioneSpesa.toString())
            .set('idProgetto', idProgetto.toString());
        return this.http.get<Array<DocumentoDiSpesaSospesoDTO>>(url, { params: params });
    }

    getDocumentiDiSpesaIntegrati(idProgetto: number, idDichiarazioneSpesa: number, idIntegrazione: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getDocumentiDiSpesaIntegrati';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString())
            .set('idDichiarazioneSpesa', idDichiarazioneSpesa.toString())
            .set('idProgetto', idProgetto.toString());
        return this.http.get<Array<any>>(url, { params: params });
    }


    getQuietanze(idDocumentoDiSpesa: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getQuietanze';
        let params = new HttpParams().set('idDocumentoSpesa', idDocumentoDiSpesa.toString());
        return this.http.get<Array<QuietanzaDTO>>(url, { params: params });
    }

    getAllegatiQuietanza(idQuietanza: number, idDichiarazioneSpesa: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegatiQuietanza';
        let params = new HttpParams()
            .set('idDichiarazioneSpesa', idDichiarazioneSpesa.toString())
            .set('idQuietanza', idQuietanza.toString());
        return this.http.get<Array<any>>(url, { params: params });
    }

    getAllegatiNuovaIntegrazioneQuietanza(idQuietanza: number, idIntegrazione: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegatiNuovaIntegrazioneQuietanza';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString())
            .set('idQuietanza', idQuietanza.toString());
        return this.http.get<Array<any>>(url, { params: params });
    }

    getAllegatiIntegrazioneQuietanza(idQuietanza: number, idDichiarazioneSpesa: number, idIntegrazione: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegatiIntegrazioneQuietanza';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString())
            .set('idDichiarazioneSpesa', idDichiarazioneSpesa.toString())
            .set('idQuietanza', idQuietanza.toString());
        return this.http.get<Array<any>>(url, { params: params });
    }

    getAllegatiIntegrazioneDocS(idDocumentoDiSpesa: number, idDichiarazioneSpesa: number, idIntegrazione: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegatiIntegrazioneDocS';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString())
            .set('idDichiarazioneSpesa', idDichiarazioneSpesa.toString())
            .set('idDocumentoDiSpesa', idDocumentoDiSpesa.toString());
        return this.http.get<Array<any>>(url, { params: params });
    }

    getAllegatiNuovaIntegrazioneDocS(idDocumentoDiSpesa: number, idIntegrazione: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegatiNuovaIntegrazioneDocS';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString())
            .set('idDocumentoDiSpesa', idDocumentoDiSpesa.toString());
        return this.http.get<Array<any>>(url, { params: params });
    }

    getAllegatiDocumentoSpesaDocS(idDocumentoDiSpesa: number, idDichiarazioneSpesa: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegatiDocumentoSpesa';
        let params = new HttpParams()
            .set('idDichiarazioneSpesa', idDichiarazioneSpesa.toString())
            .set('idDocumentoDiSpesa', idDocumentoDiSpesa.toString());
        return this.http.get<Array<any>>(url, { params: params });
    }


    getReportValidazione(idIntegrazione: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getReportValidazione';
        let params = new HttpParams().set('idIntegrazione', idIntegrazione.toString());
        return this.http.get<Array<any>>(url, { params: params });
    }

    deleteAllegati(allegato) {
        let url = this.configService.getApiURL() + `/restfacade/gestioneIntegrazioni/deleteAllegato`;
        return this.http.post<any>(url, allegato)
    }

    rimuoviAllegato(idFileEntita) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/rimuoviAllegato';
        let params = new HttpParams()
            .set("idFileEntita", idFileEntita.toString())
        return this.http.delete<boolean>(url, { params: params });
    }



    salvaAllegati(allegati) {
        let url = this.configService.getApiURL() + `/restfacade/gestioneIntegrazioni/salvaAllegati`;
        return this.http.post<any>(url, allegati)
    }

    salvaAllegatiGenerici(allegati: Array<AllegatiDichiarazioneSpesaVO>, idDichiarazioneSpesa: number, idIntegrazioneSpesa: number) {
        let url = this.configService.getApiURL() + `/restfacade/gestioneIntegrazioni/salvaAllegatiGenerici`;
        let params = new HttpParams().set('idDichiarazioneSpesa', idDichiarazioneSpesa.toString())
            .set('idIntegrazioneSpesa', idIntegrazioneSpesa.toString());
        return this.http.post<any>(url, allegati, { params: params })
    }

    inviaIntegrazioneRendicontazione(idIntegrazione: number) {
        let url = this.configService.getApiURL() + `/restfacade/gestioneIntegrazioni/inviaIntegrazioneRendicontazione/${idIntegrazione}`;
        return this.http.post<any>(url, idIntegrazione)

    }

    //INTEGRAZIONE REVOCHE

    getIntegrazioniRevoca(idProgetto: number, idBandoLinea: number): Observable<Array<IntegrazioniRevocaDTO>> {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getIntegrazioniRevoca';
        let params = new HttpParams()
            .set('idProgetto', idProgetto.toString())
            .set('idBandoLinea', idBandoLinea.toString());
        return this.http.get<Array<IntegrazioniRevocaDTO>>(url, { params: params });
    }


    uploadFileIntegrazioni(datiBackend: AllegatiControlliVO[], idRichIntegrazione: number) {
        let url = this.configService.getApiURL() + `/restfacade/gestioneIntegrazioni/salvaUploadLetteraAllegatoIntegr/${idRichIntegrazione}`;
        return this.http.post<any>(url, datiBackend)

    }

    aggiornaIntegrazione(datiBackend) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/aggiornaIntegrazione';
        let headers = new HttpHeaders().set('Content-Type', 'application/json');
        return this.http.post<any>(url, datiBackend, { headers })

    }

    getAllegatiIntegrazione(idRichIntegraz) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getAllegati';
        let params = new HttpParams().set('idRichIntegraz', idRichIntegraz.toString());
        return this.http.get<any[]>(url, { params: params });
    }

    getLetteraIstruttore(idIntegrazione) {
        let url = this.configService.getApiURL() + '/restfacade/IntegrazioniControlliInLoco/getLetteraIstruttore';
        let params = new HttpParams().set('idIntegrazione', idIntegrazione.toString());
        return this.http.get<any[]>(url, { params: params });
    }


    aggiornaIntegrazioneRevoche(idIntegrazione: number) {
        let url = this.configService.getApiURL() + `/restfacade/gestioneIntegrazioni/inviaIntegrazione/${idIntegrazione}`;
        return this.http.post<any>(url, idIntegrazione)

    }

    //Controlli in loco

    getAllegatiControlli(idIntegrazione) {
        let url = this.configService.getApiURL() + '/restfacade/IntegrazioniControlliInLoco/getAllegati';
        let params = new HttpParams().set('idIntegrazione', idIntegrazione.toString());
        return this.http.get<any[]>(url, { params: params });
    }

    uploadFileControlli(datiBackend: AllegatiControlliVO[], idIntegrazione: number) {
        let url = this.configService.getApiURL() + `/restfacade/IntegrazioniControlliInLoco/inserisciLetteraAllegato/${idIntegrazione}`;
        return this.http.post<any>(url, datiBackend)

    }

    getIntegrazioniControlli(idProgetto: number): Observable<Array<IntegrazioniRevocaDTO>> {
        let url = this.configService.getApiURL() + '/restfacade/IntegrazioniControlliInLoco/getIntegrazioniList';
        let params = new HttpParams()
            .set('idProgetto', idProgetto.toString());
        return this.http.get<Array<IntegrazioniControlliDTO>>(url, { params: params });
    }

    /* aggiornaControlli(idIntegrazione: number): Observable<Array<IntegrazioniRevocaDTO>> {
        let url = this.configService.getApiURL() + '/restfacade/IntegrazioniControlliInLoco/aggiornaIntegrazione';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString());
        return this.http.get<Array<IntegrazioniControlliDTO>>(url, { params: params });
    }
 */
    aggiornaControlli(idIntegrazione: number) {
        let url = this.configService.getApiURL() + `/restfacade/IntegrazioniControlliInLoco/aggiornaIntegrazione/${idIntegrazione}`;
        return this.http.post<any>(url, idIntegrazione)

    }



    //PROREGHE 

    getInfoProrogheControlli(idIntegrazione: number, idControllo: number, idProgetto: number): Observable<Array<IntegrazioniRevocaDTO>> {
        let url = this.configService.getApiURL() + '/restfacade/IntegrazioniControlliInLoco/getAbilitaRichProroga';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString())
            .set('idControllo', idControllo.toString())
            .set('idProgetto', idProgetto.toString());
        return this.http.get<Array<IntegrazioniRevocaDTO>>(url, { params: params });
    }

    getInfoProrogheRevoche(idIntegrazione: number): Observable<Array<IntegrazioniRevocaDTO>> {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getRichProroga';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString());
        return this.http.get<Array<IntegrazioniRevocaDTO>>(url, { params: params });
    }

    getInfoProrogheRendicontazione(idIntegrazione: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/getRichProrogaRendicontazione';
        let params = new HttpParams()
            .set('idIntegrazione', idIntegrazione.toString());
        return this.http.get<Array<IntegrazioniRevocaDTO>>(url, { params: params });

    }


    inserisciRichProrogaControlli(datiBackend, idIntegrazione: number) {
        let url = this.configService.getApiURL() + `/restfacade/IntegrazioniControlliInLoco/inserisciRichProroga/${idIntegrazione}`;
        return this.http.post<any>(url, datiBackend)

    }

    inserisciRichProrogaRendicontazione(datiBackend, idIntegrazione: number) {
        let url = this.configService.getApiURL() + `/restfacade/gestioneIntegrazioni/inserisciRichProrogaRendicontazione/${idIntegrazione}`;
        return this.http.post<any>(url, datiBackend)

    }

    inserisciRichProrogaRevoche(datiBackend, idIntegrazione: number) {
        let url = this.configService.getApiURL() + `/restfacade/gestioneIntegrazioni/inserisciRichProroga/${idIntegrazione}`;
        return this.http.post<any>(url, datiBackend)

    }

    getAllegatiDocSpesaQuietanzeDaInviare(idIntegrazioneSpesa: number) {
        let url = this.configService.getApiURL() + '/restfacade/gestioneIntegrazioni/allegatiDocSpesaQuietanzeDaInviare';
        let params = new HttpParams().set('idIntegrazioneSpesa', idIntegrazioneSpesa.toString());
        return this.http.get<Array<AllegatiDocSpesaQuietanzeDTO>>(url, { params: params });

    }
}
